src/VisualStudio/TestUtilities/Window/EditorWindow.cs

```diff
@@ -1,30 +1,279 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
+using System.Reflection;
+using EnvDTE;
+using Roslyn.VisualStudio.Test.Utilities.Interop;
 using Roslyn.VisualStudio.Test.Utilities.Remoting;
 
+using Thread = System.Threading.Thread;
+
 namespace Roslyn.VisualStudio.Test.Utilities
 {
     /// <summary>Provides a means of interacting with the active editor window in the Visual Studio host.</summary>
     public class EditorWindow
     {
-        private readonly VisualStudioInstance _visualStudio;
+        private readonly VisualStudioInstance _visualStudioInstance;
+        private readonly EditorWindowWrapper _editorWindowWrapper;
+
+        internal EditorWindow(VisualStudioInstance visualStudioInstance)
+        {
+            _visualStudioInstance = visualStudioInstance;
+
+            var integrationService = _visualStudioInstance.IntegrationService;
+            _editorWindowWrapper = integrationService.Execute<EditorWindowWrapper>(typeof(EditorWindowWrapper), nameof(EditorWindowWrapper.Create), (BindingFlags.Public | BindingFlags.Static));
+        }
+
+        public int CurrentColumn
+        {
+            get
+            {
+                return 0;
+            }
+        }
 
-        internal EditorWindow(VisualStudioInstance visualStudio)
+        public string CurrentLineTextBeforeCursor
         {
-            _visualStudio = visualStudio;
+            get
+            {
+                // Select everything from the cursor to the beginning of the line
+                TextSelection.StartOfLine(vsStartOfLineOptions.vsStartOfLineOptionsFirstColumn, Extend: true);
+
+                var currentLineTextBeforeCursor = TextSelection.Text;
+
+                // Restore the cursor to its previous position
+                if (currentLineTextBeforeCursor != string.Empty)
+                {
+                    TextSelection.CharRight(false, 1);
+                }
+
+                return currentLineTextBeforeCursor;
+            }
+        }
+
+        public string CurrentLineTextAfterCursor
+        {
+            get
+            {
+                // Clear the current selected text, if any, so it is not included in the returned value
```

Comment:
Why is this not done in the before cursor function above? Keep in mind the anchor point for a selection can be either way.